### PR TITLE
feat: add flag ui-port

### PR DESCRIPTION
**Pull request by FlorianKutz**
**What issue type does this pull request address?** (keep at least one, remove the others)  
/kind feature

**What is this pull request for? Which issues does it resolve?** (use `resolves #<issue_number>` if possible)  
it adds a new flag

**Does this pull request has user-facing changes?** (e.g. config changes, new/modified commands, new/modified flags)  
yes, the user can now use a new flag when running devspace dev

**Does this pull request add new dependencies?**  
no

**What else do we need to know?**

### CODE REVIEW

File: cmd/dev.go [hunk 1]
```diff
@@ -399,13 +402,19 @@ func (cmd *DevCmd) startServices(f factory.Factory, config *latest.Config, gener
 	}
 
 	// Open UI if configured
-	if cmd.UI {
+	if cmd.UI || cmd.UIPort != 0 {
```
Comment (FabianKramm) [inline cmd/dev.go:405]: We should stay with only `cmd.UI` check here

File: cmd/dev.go [hunk 2]
```diff
@@ -399,13 +402,19 @@ func (cmd *DevCmd) startServices(f factory.Factory, config *latest.Config, gener
...
 		cmd.UI = false
 		logger.StartWait("Starting the ui server...")
 		defer logger.StopWait()
 
+		var port *int
+		if cmd.UIPort != 0 {
+			port = ptr.Int(cmd.UIPort)
+		}
+		cmd.UIPort = 0
```
Comment (FabianKramm) [inline cmd/dev.go:414]: why is that necessary?

File: cmd/dev.go [hunk 3]
```diff
@@ -399,13 +402,19 @@ func (cmd *DevCmd) startServices(f factory.Factory, config *latest.Config, gener
...
 	// Open UI if configured
-	if cmd.UI {
+	if cmd.UI || cmd.UIPort != 0 {
 		cmd.UI = false
 		logger.StartWait("Starting the ui server...")
 		defer logger.StopWait()
 
+		var port *int
+		if cmd.UIPort != 0 {
+			port = ptr.Int(cmd.UIPort)
```
Comment (FabianKramm) [inline cmd/dev.go:412]: why not just use &cmd.UIPort?

**Review by FabianKramm — CHANGES_REQUESTED** [review]

File: cmd/dev.go (hunk 3 above)
Comment (FlorianKutz) [reply cmd/dev.go:412]: Because otherwise whenever cmd.UIPort == 0 the parameter forcePort isn't nil. It would be a pointer to 0, which forces the port 0 to be used.

File: cmd/dev.go (hunk 2 above)
Comment (FlorianKutz) [reply cmd/dev.go]: It was to prevent the server from starting twice. Because of another change, it's not needed anymore.
